src/main/java/com/google/sps/servlets/ChatServlet.java

```diff
@@ -24,35 +26,87 @@
-    String chatid = request.getParameter("chatid");
-    String userid = request.getParameter("participantid");
-    response.getWriter().println("Chat servlet working\n");
-    response.getWriter().println("Endpoint: " + endpoint);
-    response.getWriter().println("ChatID: " + chatid);
-    response.getWriter().println("UserID: " + userid);
     Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+    KeyFactory keyFactory = datastore.newKeyFactory();
 
-    Gson gson = new Gson();
-    //Get all chats that belong to the user to put in the menu
-    if (userid != null){
-        List<Chat> chats = Chat.getChats(datastore, userid);
-        response.getWriter().println(gson.toJson(chats));
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String endpoint = request.getPathInfo();
+        String chatid = request.getParameter("chatid");
+        String userid = request.getParameter("userid");
+        response.getWriter().println("Chat servlet working\n");
+        response.getWriter().println("Endpoint: " + endpoint);
+        Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+
+        Gson gson = new Gson();
+        
+        //Get a specific chat if the chat id is passed
+        if (chatid != null){
+            Chat chat = Chat.getUserChat(datastore, chatid);
+            response.getWriter().println("Chat " + chatid + " - Messages: " + chat.getMessages());
+            response.getWriter().println(gson.toJson(chat));
+        }
+        //Get all chats that belong to the user to put in the menu
+        else if (userid != null){
+            List<Chat> chats = Chat.getUserChats(datastore, userid);
+            response.getWriter().println("Number of chats belonging to " + userid + " = " + chats.size());
+            response.getWriter().println(gson.toJson(chats));
+        }
+        else{
+            List<Chat> chats = Chat.getAllChats(datastore);
+            response.getWriter().println("Number of all chats = " + chats.size());
+            response.getWriter().println(gson.toJson(chats));
+        }
+        // OBJECTIVE: Create a chat between users
+        // response.sendRedirect("dashboard.html");
     }
-    //Get a specific chat if the chat id is passed
-    else if (chatid != null){
-        Chat chat = Chat.getChat(datastore, userid);
-        response.getWriter().println(gson.toJson(chat));
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException { // create
+
+      // Get the user id and the partner id that they're matching with
+    //   String userid = Jsoup.clean(request.getParameter("id"), Whitelist.none());
+      String userid = "115545090202496645724"; // Test id 1
+    //   String partnerid = Jsoup.clean(request.getParameter("partnerid"), Whitelist.none());
+      String partnerid = "104737820262548186234"; //Test id 2
```

Comment:
[Local variable names are written in lowerCamelCase.](https://google.github.io/styleguide/javaguide.html#s5.2.7-local-variable-names)

```suggestion
      String partnerId = "104737820262548186234"; // Test id 2
```